pontoon/base/tests/models/test_stats.py

```diff
@@ -0,0 +1,162 @@
+"""
+Test consistency of calculations between `calculate_stats` and `translation.save()`.
+"""
+import pytest
+from pontoon.base.models import TranslatedResource
+from pontoon.checks.models import (
+    Error,
+    Warning,
+)
+
+
+@pytest.fixture
+def translation_with_error(translation_a):
+    Error.objects.create(
+        translation=translation_a,
+        library='p',
+        message='error'
+    )
+    return translation_a
+
+
+@pytest.fixture
+def translation_with_warning(translation_a):
+    Warning.objects.create(
+        translation=translation_a,
+        library='p',
+        message='warning',
+    )
+    return translation_a
+
+
+def recalculate_stats(translation):
+    """
+    Make the full recalculate stats on a TranslatedResource.
+    """
+    translation.save(update_stats=False)
+    translated_resource = TranslatedResource.objects.get(
+        resource=translation.entity.resource,
+        locale=translation.locale,
+    )
+    translated_resource.calculate_stats()
+
+
+def diff_stats(t):
+    """
+    Update only necessary stats by calculating difference between stats.
+    """
+    t.save()
+
+
+@pytest.fixture(
+    params=(
+        recalculate_stats,
+        diff_stats,
+    )
+)
+def stats_update(db, request):
+    """
+    Wrapper fixture which allows to test both implementations of stats calculations.
+    """
+    return request.param
```

Comment:
I'm curious, how does that work. The 1st call of `stats_update` calls `recalculate_stats` and the 2nd calls `diff_stats`?

Reply:
Exactly, this will test both types of calculations.

Reply:
Fancy! :)